hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/CryptoCreateTransitionLogic.java

```diff
@@ -244,10 +236,15 @@ public ResponseCodeEnum validate(TransactionBody cryptoCreateTxn) {
         return cryptoCreateChecks.cryptoCreateValidation(cryptoCreateTxn.getCryptoCreateAccount());
     }
 
-    private void populateKeyAndAliasInCaseOfAliasOrAliasAndEvmAddressProvided(
+    private void populateKeyAndAliasInCaseOfAliasProvided(
             final CryptoCreateTransactionBody op, final HederaAccountCustomizer customizer) {
-        final var keyFromAlias = asPrimitiveKeyUnchecked(op.getAlias());
-        final JKey jKeyFromAlias = asFcKeyUnchecked(keyFromAlias);
-        customizer.key(jKeyFromAlias).alias(op.getAlias());
+        if (op.getAlias().size() != EVM_ADDRESS_SIZE) {
```

Comment:
By this point in the review its becoming clear to me that something needs to be refactored here.  Either a:
- one line `isEvmAddress()` method, or
- a `addressKind()` method returning an enum (EVM, EDCSA, LONGZERO, whatever).

Consider just locally here:  at line 241 we have `!= EVM_ADDRESS_SIZE` and just below at line 245 we have `== EVM_ADDRESS_SIZE`.  Wouldn't something self-explanatory (by name) be better?  Or - a switch on address type (that itself would be a further benefit when later we need to make changes for some existing or new address type:  Look for the switches on that method rather than looking for `!= EVM_ADDRESS_SIZE`.)

Reply:
Could even include the null (empty/missing field) check here, i.e., within the new method - if the switch `addressKind` is chosen there could be an `INVALID_ADDRESS` or it could return an `Optional` (I do like using `Optional` to make null-checks explicit in the code but I gotta admit that in current Java the syntax at sites _consuming_ an `Optional` is kind of clunky.)  See at (new) line 120.